Update API documentation

kybowm opened the pull request:
There are two commits in this PR. The first adds the API changes to the documentation. The second reorganizes the API into groups for Annotaitons, Tables, Utilities, and Visualization.

Here is a preview of the doc: https://kybowm.github.io/python-cvpy/api.html#

In a follow up, I want to get the table of contents to expand for each level of the API.

ptelang commented on doc/source/api.rst:
```diff
@@ -5,66 +5,108 @@
 
-CAS Thread Tuner
-----------------
+Project
```
Since Project is a base class, we don't expect end users to ever create or interact with this class. So, we can completely remove it from the documentation.

ptelang commented on doc/source/api.rst:
```diff
@@ -5,66 +5,108 @@
+    CVATProject.post_images
+    CVATProject.save
+    CVATProject.resume
+    CVATProject.to_json
```
The end users should never need to use as_dict, from_json and to_json functions. So these can be removed.

ptelang commented on doc/source/api.rst:
```diff
@@ -5,66 +5,108 @@
+    CVATProject.resume
+    CVATProject.to_json
+
+CVAT Task
```
The end users should never have to create or interact with a CVATTask object. It is automatically created by the CVATProject.post_images() call. So it can be completely removed from the doc.

ptelang commented on doc/source/api.rst:
```diff
@@ -5,66 +5,108 @@
-    Image.get_image_array_const_ctype
-    Image.mask_image
+    CVATAuthenticator
+    CVATAuthenticator.authenticate
```
The end users should never use CVATAuthenticator.authenticate function. They are expected to only use generate_cvat_token. So we can remove CVATAuthenticator.authenticate from the doc.

ptelang commented on doc/source/api.rst:
```diff
@@ -5,66 +5,108 @@
+    ImageTable.has_decoded_images
+    BiomedImageTable.load
+    BiomedImageTable.sphericity
+    BiomedImageTable.morphological_gradient
```
We expect users to call ImageTable.from_table() and ImageTable.load() functions. So we can remove the following. 
BiomedImageTable.from_table
BiomedImageTable.load

ptelang commented on doc/source/api.rst:
```diff
@@ -74,55 +116,77 @@ ImageTable
+    NaturalImageTable.has_decoded_images
+    NaturalImageTable.from_table
+    NaturalImageTable.load
+    NaturalImageTable.mask_image
```
We expect users to call ImageTable.from_table() and ImageTable.load() functions. So we can remove the following.
NaturalImageTable.from_table
NaturalImageTable.load

ptelang reviewed (commented):
Hi Kyle, Thanks for updating the API doc. I have a few comments about removing some functions from the doc.

kybowm commented:
@ptelang , thank you for your feedback! I've incorporated your suggestions. I also saw a couple of ImageTable methods in the BiomedImageTable section and replaced them with their BiomedImageTable counterparts.

Updates are posted to the same URL: https://kybowm.github.io/python-cvpy/api.html#
